### PR TITLE
Add monkey patch for Qwen2 models

### PR DESCRIPTION
**Pull request by chiwanpark**
## Summary
This PR adds a monkey patch script for Qwen2 models. I've copied the monkey patch script for Mixtral because the Qwen2 models have the same structure of Mixtral. Also, I've checked the `forward` method of `Qwen2ForCausalLM` class ([link](https://github.com/huggingface/transformers/blob/main/src/transformers/models/qwen2/modeling_qwen2.py#L1058)) to apply the fused linear cross entropy loss.

## Testing Done

- [x] run `make test` to ensure correctness
- [x] run `make checkstyle` to ensure code style
- [x] run `make test-convergence` to ensure convergence

```zsh
❯ make test
pytest --disable-warnings test/ --ignore=test/convergence
================================================ test session starts =================================================
platform linux -- Python 3.10.12, pytest-8.3.2, pluggy-1.5.0
rootdir: /data/ovis_915/bell/Liger-Kernel.git
plugins: anyio-4.4.0
collected 114 items                                                                                                  

test/transformers/test_cross_entropy.py ..........................................................             [ 50%]
test/transformers/test_fused_linear_cross_entropy.py ......                                                    [ 56%]
test/transformers/test_geglu.py ........                                                                       [ 63%]
test/transformers/test_rms_norm.py ................                                                            [ 77%]
test/transformers/test_rope.py ............                                                                    [ 87%]
test/transformers/test_swiglu.py ........                                                                      [ 94%]
test/transformers/test_trainer_integration.py ...                                                              [ 97%]
test/transformers/test_transformers_monkey_patch.py .                                                          [ 98%]
test/triton/test_triton_monkey_patch.py ..                                                                     [100%]

=========================================== 114 passed in 62.15s (0:01:02) ===========================================
```

```zsh
❯ make checkstyle 
flake8 .; flake8_status=$?; \
isort .; isort_status=$?; \
black .; black_status=$?; \
if [ $flake8_status -ne 0 ] || [ $isort_status -ne 0 ] || [ $black_status -ne 0 ]; then \
        exit 1; \
fi
Skipped 1 files
All done! ✨ 🍰 ✨
52 files left unchanged.
```

```zsh
❯ make test-convergence 
HF_DATASETS_OFFLINE=1 pytest --disable-warnings test/convergence
================================================ test session starts =================================================
platform linux -- Python 3.10.12, pytest-8.3.2, pluggy-1.5.0
rootdir: /data/ovis_915/bell/Liger-Kernel.git
plugins: anyio-4.4.0
collected 8 items                                                                                                    

test/convergence/test_mini_models.py ......                                                                    [ 75%]
test/convergence/test_mini_models_no_logits.py ..                                                              [100%]

================================================= 8 passed in 55.44s =================================================

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -152,6 +152,7 @@ loss.backward()
 | Mistral     | `liger_kernel.transformers.apply_liger_kernel_to_mistral`  | RoPE, RMSNorm, SwiGLU, CrossEntropyLoss        |
 | Mixtral     | `liger_kernel.transformers.apply_liger_kernel_to_mixtral`  | RoPE, RMSNorm, SwiGLU, CrossEntropyLoss        |
 | Gemma2      | `liger_kernel.transformers.apply_liger_kernel_to_gemma`    | RoPE, RMSNorm, GeGLU, CrossEntropyLoss         |
+| Qwen2       | `liger_kernel.transformers.apply_liger_kernel_to_qwen2`    | RoPE, RMSNorm, SwiGLU, CrossEntropyLoss        |
```
Comment (ByronHsu) [inline README.md:155]:
```suggestion
| Qwen2       | `liger_kernel.transformers.apply_liger_kernel_to_qwen2`    | RoPE, RMSNorm, SwiGLU, CrossEntropyLoss, FusedLinearCrossEntropy        |

```

**Review by ByronHsu — COMMENTED** [review]
This shipping speed is crazy!

**Review by ByronHsu — APPROVED** [review]
